src/reducers/clusterReducer.js

```diff
@@ -108,16 +73,9 @@ export default function clusterReducer(
       };
 
     case types.CLUSTER_LOAD_DETAILS_SUCCESS:
-      items = Object.assign({}, state.items);
-
-      items[action.cluster.id] = Object.assign(
-        {},
-        items[action.cluster.id],
-        action.cluster
-      );
+      items = Object.assign(state.items, {});
```

Comment:
My understanding is that when we do this we actually mutate the `state.items` object, instead of creating a new object. which is an anti pattern (mutating state). 

I think things are actually alright in this action (CLUSTER_LOAD_DETAILS_SUCCESS), did you find these changes necessary?

Reply:
Good question. I didn't make sure. Will revert this part and see how that behaves.